### PR TITLE
System.Attribute F# snippets

### PR DESCRIPTION
**Pull request by albert-du**
Adds F# snippets for the Attribute class

### CODE REVIEW

File: samples/snippets/fsharp/VS_Snippets_CLR/AnimalAttributes/FS/customattribute.fs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    member _.DogMethod() = ()
+
+    [<AnimalType(Animal.Cat)>]
+    member _.CatMethod() = ()
+
+    [<AnimalType(Animal.Bird)>]
+    member _.BirdMethod() = ()
+
+let testClass = AnimalTypeTestClass()
+let type' = testClass.GetType()
```
Comment (dsyme) [inline samples/snippets/fsharp/VS_Snippets_CLR/AnimalAttributes/FS/customattribute.fs:27]: Please don't use names ending with `'` - change to `theClassType` or `theType` or `typ`

**Review by dsyme — APPROVED** [review]
I'll mark this as approved though I left one small change request

File: samples/snippets/fsharp/VS_Snippets_CLR/GetCustomAttributes/FS/ca2.fs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+// Set the module's CLSCompliant attribute to false
+// The CLSCompliant attribute is applicable for /target:module.
+[<``module``: CLSCompliant false>]
+do ()
+
+type DemoClass = class end
+
+// Get the Module type to access its metadata.
+let module' = typeof<DemoClass>.Module
```
Comment (BillWagner) [inline samples/snippets/fsharp/VS_Snippets_CLR/GetCustomAttributes/FS/ca2.fs:18]: Same comment: the variable should not end in `'`:

```suggestion
let module = typeof<DemoClass>.Module
```

File: samples/snippets/fsharp/VS_Snippets_CLR/GetCustomAttributes/FS/ca2.fs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+// Iterate through all the attributes for the module.
+for attr in Attribute.GetCustomAttributes module' do
```
Comment (BillWagner) [inline samples/snippets/fsharp/VS_Snippets_CLR/GetCustomAttributes/FS/ca2.fs:21]: change variable name:

```suggestion
for attr in Attribute.GetCustomAttributes module do
```

File: samples/snippets/fsharp/VS_Snippets_CLR/GetCustomAttributes/FS/ca2.fs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    match attr with
+    // Check for the Description attribute.
+    | :? DescriptionAttribute as attr ->
+        printfn $"Module {module'.Name} has the description \"{attr.Description}\"."
+                
+    // Check for the CLSCompliant attribute.
+    | :? CLSCompliantAttribute as attr ->
+        printfn $"""Module {module'.Name} {if attr.IsCompliant then "is" else "is not"} CLSCompliant."""
```
Comment (BillWagner) [inline samples/snippets/fsharp/VS_Snippets_CLR/GetCustomAttributes/FS/ca2.fs:29]: variable name change:

```suggestion
        printfn $"Module {module.Name} has the description \"{attr.Description}\"."
                
    // Check for the CLSCompliant attribute.
    | :? CLSCompliantAttribute as attr ->
        printfn $"""Module {module.Name} {if attr.IsCompliant then "is" else "is not"} CLSCompliant."""
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks again @albert-du 

I found one more instance of a variable with `'` at the end. I made a suggestion for those.  Once you accept that suggestion, and the build is clean, I'll :shipit:.

Thanks again for all you're doing here to help us improve!

File: samples/snippets/fsharp/VS_Snippets_CLR/GetCustomAttributes/FS/ca2.fs (hunk 2 above)
Comment (albert-du) [reply samples/snippets/fsharp/VS_Snippets_CLR/GetCustomAttributes/FS/ca2.fs:18]: module is a reserved keyword, I'll change it something else.

**Review by BillWagner — APPROVED** [review]

**Pull request merged**